client/signup/steps/domains/index.jsx

```diff
@@ -896,10 +940,12 @@ export class RenderDomainsStep extends Component {
 						</div>
 						<div key="rowtotalprice" className="domains__domain-cart-total-price">
 							<strong>
-								{ formatCurrency(
-									domainsInCart.reduce( ( total, item ) => total + item.cost, 0 ),
-									domainsInCart.length ? domainsInCart[ 0 ].currency : 'USD'
-								) }
+								{ domainsInCart.some( ( domain ) => domain.temporary )
```

Comment:
👋 For unrelated reasons I'm trying to grok what goes on in this class. From what I can tell, `domainsInCart` is an array of shopping cart items, but `temporary` is not a shopping cart item property. Where does this property get added? (Currently it has been moved to `domains-mini-cart.jsx` but the code here is the same.)

Reply:
`temporary` gets set here https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/domains/index.jsx#L629, some shopping cart endpoint calls take multiple seconds to respond so this temporary cart approach was added client side to make the user experience feel snappier while we wait for the server side to catch up.

Reply:
Hey @sirbrillig, we add the `temporary` property [in the addDomain handler](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/domains/index.jsx#L622-L632) inside the `temporaryCart`only. Its goal is to have a temporary cart while we sync the items with the backend so we can instantly show to the user that domain in the mini-cart.

Reply:
Had a cached page open an didn't see your comment before sending mine @lsl. Thanks for replying to it!

Reply:
Got it. Thank you for clarifying! That's a clever approach.